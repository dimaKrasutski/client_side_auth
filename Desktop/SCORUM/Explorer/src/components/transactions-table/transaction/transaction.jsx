import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import Table from 'reactstrap/lib/Table';
import { IconPaymentConvertArrow } from 'theme/components/icons/icon-payment/icon-payment-convert-arrow';
import { RouteLink } from '../../../components/route-link/route-link';
import { formatBalanceWithLocale } from '../../../helpers/accounts';
import { openModal } from '../../../actions/modals';
import { ModalWindow } from '../../modal-window/modal-window';

class TransactionUI extends Component {
  static propTypes = {
    openDetailsModal: PropTypes.func.isRequired,
    locale: PropTypes.string.isRequired,
    hash: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    height: PropTypes.number.isRequired,
    transaction: PropTypes.shape({
      timestamp: PropTypes.string.isRequired,
      block_id: PropTypes.string.isRequired,
      transactions: PropTypes.shape({
        amount: PropTypes.string.isRequired,
        from: PropTypes.string.isRequired,
        to: PropTypes.string.isRequired,
        memo: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  };

  openDetailsModalHandler = (e) => {
    if (e.target.tagName !== 'A') {
      this.props.openDetailsModal(this.props.hash);
    }
  };

  render() {
    const {
      locale,
      transaction,
      hash,
      height,
      date,
    } = this.props;

    const formatWithLocale = formatBalanceWithLocale(locale);
    const {
      // TODO: remove
      operations: [[, {
        amount,
        from,
        to,
        memo,
      }]],
    } = transaction;

    return (
      <Fragment>
        <tr className="transactions" onClick={this.openDetailsModalHandler}>
          <td>
            {hash}
          </td>
          <td>
            <RouteLink to={`/account-info/${from}`} locale={locale}>
              {from}
            </RouteLink>
          </td>
          <td>
            <IconPaymentConvertArrow />
          </td>
          <td>
            <RouteLink to={`/account-info/${from}`} locale={locale}>
              {to}
            </RouteLink>
          </td>
          <td>
            <div className="date">{date}</div>
          </td>
          <td className="digit-value amount">
            {formatWithLocale(amount)}
          </td>
        </tr>
        <ModalWindow
          id={hash}
          bodyClassName="transaction-details"
          size="lg"
        >
          <h2 className="mb-3">
            <FormattedMessage id="explorer.tr-details.title" />
          </h2>
          <Table>
            <tbody>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.date" /></td>
                <td>{date}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.amount" /></td>
                <td>{amount}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.hash" /></td>
                <td>{hash}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.to" /></td>
                <td>{to}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.memo" /></td>
                <td>{memo}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.from" /></td>
                <td>{from}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.block" /></td>
                <td>{height}</td>
              </tr>
            </tbody>
          </Table>
        </ModalWindow>
      </Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  openDetailsModal: id => dispatch(openModal(id)),
});

export const Transaction = connect(null, mapDispatchToProps)(TransactionUI);
