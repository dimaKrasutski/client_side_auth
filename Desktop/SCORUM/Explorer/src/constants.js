import * as THEME_CONSTANTS from 'theme/constants';

export const {
  LOCALE_KEYS,
  LANG_KEYS,
  LOCALE,
  LOCALES,
  DEFAULT_LOCALE,
  DEFAULT_LANG,
} = THEME_CONSTANTS;

export const UNDEFINED_ERROR_ID = 'explorer.error.unknown';

export const MIN_FRACTION_DIGITS = 3;
export const MAX_FRACTION_DIGITS = 9;

export const CURRENCY_USD = 'USD';
export const CURRENCY_EUR = 'EUR';
export const ALL_CURRENCY = [CURRENCY_USD, CURRENCY_EUR];
