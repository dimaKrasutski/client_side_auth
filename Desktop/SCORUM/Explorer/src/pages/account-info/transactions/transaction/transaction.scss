.transactions {
  cursor: pointer;

  .amount {
    vertical-align: middle;
  }
}

.transaction-details {
  .table {
    tr {
      &:last-child {
        td {
          padding: 0;
        }
      }
    }
    td {
      text-align: left;
      padding: 0 0 1.25rem 0;
      border: 0;

      &:first-of-type {
        color: $gray-dark-secondary;
        white-space: nowrap;
      }

      &:last-of-type {
        word-break: break-all;
        font-weight: bold;
        max-width: 270px;
      }
    }
  }
}
