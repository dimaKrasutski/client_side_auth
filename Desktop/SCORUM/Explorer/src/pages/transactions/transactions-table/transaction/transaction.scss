.transactions-table {
  .td-overflow {
    width: 300px;
    max-width: 300px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  .user-link {
    a:hover {
      text-decoration: none;
      color: $gray-dark-secondary;
    }
  }
}
