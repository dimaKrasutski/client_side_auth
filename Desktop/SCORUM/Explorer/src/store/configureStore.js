import { createStore, applyMiddleware, compose } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import { rootEpic } from '../epics/root';
import rootReducer from '../reducers/index';

const epicMiddleware = createEpicMiddleware(rootEpic);

export default function configureStore(initialState) {
  const middlewares = [
    epicMiddleware,
  ];
  if (process.env.PROJECT_ENV !== 'production') {
    // eslint-disable-next-line
    const { logger } = require(`redux-logger`);
    middlewares.push(logger);
  }

  const enhancer = compose(applyMiddleware(...middlewares));

  return createStore(rootReducer, initialState, enhancer);
}
