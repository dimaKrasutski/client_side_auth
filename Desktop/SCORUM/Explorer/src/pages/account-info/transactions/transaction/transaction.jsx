import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import Table from 'reactstrap/lib/Table';
import { IconOutcome } from 'theme/components/icons/icon-income/icon-outcome';
import { IconIncome } from 'theme/components/icons/icon-income/icon-income';
import { toStandardFormat } from '../../../../helpers/date';
import { formatBalanceWithLocale } from '../../../../helpers/accounts';
import { LOCALE_KEYS } from '../../../../constants';
import { openModal } from '../../../../actions/modals';
import { ModalWindow } from '../../../../components/modal-window/modal-window';
import { RouteLink } from '../../../../components/route-link/route-link';
import './transaction.scss';

class TransactionUI extends Component {
  static propTypes = {
    openDetailsModal: PropTypes.func.isRequired,
    locale: PropTypes.string.isRequired,
    userName: PropTypes.string.isRequired,
    transaction: PropTypes.shape({
      timestamp: PropTypes.string.isRequired,
      trx_id: PropTypes.string.isRequired,
      block: PropTypes.number.isRequired,
      op: PropTypes.shape({
        amount: PropTypes.string.isRequired,
        from: PropTypes.string.isRequired,
        to: PropTypes.string.isRequired,
        memo: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  };

  openDetailsModalHandler = (e) => {
    if (e.target.tagName !== 'A') {
      this.props.openDetailsModal(this.props.transaction.trx_id);
    }
  };

  render() {
    const {
      locale,
      transaction,
      userName,
    } = this.props;

    const formatWithLocale = formatBalanceWithLocale(locale);
    const {
      timestamp,
      trx_id: trxId,
      block,
      op: {
        amount,
        from,
        to,
        memo,
      },
    } = transaction;
    const income = to === userName;
    const showedUserName = income ? from : to;

    return (
      <Fragment>
        <tr className="transactions" onClick={this.openDetailsModalHandler}>
          <td>
            <span>{trxId}</span>
          </td>
          <td>
            {income ? <IconIncome className="mr-1_5" /> : <IconOutcome className="mr-1_5" />}
            <RouteLink to={`/account-info/${showedUserName}`}>
              {showedUserName}
            </RouteLink>
          </td>
          <td>
            {toStandardFormat(timestamp, locale)}
          </td>
          <td className="digit-value amount">
            {income ? '+' : '-'} {formatWithLocale(amount)} SCR
          </td>
        </tr>
        <ModalWindow
          id={trxId}
          bodyClassName="transaction-details"
          size="lg"
        >
          <h2 className="mb-3">
            <FormattedMessage id="explorer.tr-details.title" />
          </h2>
          <Table>
            <tbody>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.date" /></td>
                <td>{toStandardFormat(timestamp, LOCALE_KEYS.RU_RU)}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.amount" /></td>
                <td>{amount}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.hash" /></td>
                <td>{trxId}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.to" /></td>
                <td>{to}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.memo" /></td>
                <td>{memo}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.from" /></td>
                <td>{from}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.block" /></td>
                <td>{block}</td>
              </tr>
            </tbody>
          </Table>
        </ModalWindow>
      </Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  openDetailsModal: id => dispatch(openModal(id)),
});

export const Transaction = connect(null, mapDispatchToProps)(TransactionUI);
