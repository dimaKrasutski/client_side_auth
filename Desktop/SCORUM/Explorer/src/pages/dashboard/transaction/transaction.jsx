import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { toStandardFormat } from '../../../helpers/date';
import { LOCALE_KEYS } from '../../../constants';
import { openModal } from '../../../actions/modals';
import { ModalWindow } from '../../../components/modal-window/modal-window';

class TransactionUI extends Component {
  static propTypes = {
    openDetailsModal: PropTypes.func.isRequired,
    transaction: PropTypes.shape({
      timestamp: PropTypes.string.isRequired,
      block: PropTypes.number.isRequired,
      trx_id: PropTypes.string.isRequired,
      op: PropTypes.arrayOf(
        PropTypes.oneOfType(
          [
            PropTypes.string,
            PropTypes.shape({
              amount: PropTypes.string.isRequired,
              from: PropTypes.string.isRequired,
              to: PropTypes.string.isRequired,
              memo: PropTypes.string.isRequired,
            }),
          ]
        )
      ),
    }).isRequired,
  };

  openDetailsModalHandler = () => {
    const { trx_id: trxId } = this.props.transaction;
    this.props.openDetailsModal(`main${trxId}`);
  };

  render() {
    const {
      timestamp,
      trx_id: trxId,
      op: [
        type,
      ],
    } = this.props.transaction;

    return (
      <Fragment>
        <tr className="transactions" onClick={this.openDetailsModalHandler}>
          <td>
            {trxId}
          </td>
          <td>
            <div className="date">{toStandardFormat(timestamp, LOCALE_KEYS.RU_RU)}</div>
          </td>
          <td className="font-weight-bold text-capitalize">
            {type.toString().replace(/_/g, ' ')}
          </td>
        </tr>
        <ModalWindow
          id={`main${trxId}`}
          bodyClassName="transaction-details"
          size="lg"
        >
          <h2 className="mb-3">
            <FormattedMessage id="explorer.tr-details.title" />
          </h2>
          <div><pre>{JSON.stringify(this.props.transaction, null, 2) }</pre></div>
        </ModalWindow>
      </Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  openDetailsModal: id => dispatch(openModal(id)),
});

export const Transaction = connect(null, mapDispatchToProps)(TransactionUI);
