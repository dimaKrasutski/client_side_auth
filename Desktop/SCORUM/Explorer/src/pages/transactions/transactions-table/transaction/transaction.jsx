import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import Table from 'reactstrap/lib/Table';
import { IconPaymentConvertArrow } from 'theme/components/icons/icon-payment/icon-payment-convert-arrow';
import { RouteLink } from '../../../../components/route-link/route-link';
import { toStandardFormat } from '../../../../helpers/date';
import { LOCALE_KEYS } from '../../../../constants';
import { openModal } from '../../../../actions/modals';
import { ModalWindow } from '../../../../components/modal-window/modal-window';
import './transaction.scss';

class TransactionUI extends Component {
  static propTypes = {
    openDetailsModal: PropTypes.func.isRequired,
    transaction: PropTypes.shape({
      timestamp: PropTypes.string.isRequired,
      block: PropTypes.number.isRequired,
      trx_id: PropTypes.string.isRequired,
      op: PropTypes.arrayOf(
        PropTypes.oneOfType(
          [
            PropTypes.string,
            PropTypes.shape({
              amount: PropTypes.string.isRequired,
              from: PropTypes.string.isRequired,
              to: PropTypes.string.isRequired,
              memo: PropTypes.string.isRequired,
            }),
          ]
        )
      ),
    }).isRequired,
  };

  openDetailsModalHandler = (e) => {
    if (e.target.tagName !== 'A') {
      this.props.openDetailsModal(this.props.transaction.trx_id);
    }
  };

  render() {
    const {
      timestamp,
      block,
      trx_id: trxId,
      // TODO: remove
      op: [, {
        amount,
        from,
        to,
        memo,
      }],
    } = this.props.transaction;

    return (
      <Fragment>
        <tr className="transactions" onClick={this.openDetailsModalHandler}>
          <td className="td-overflow">
            {trxId}
          </td>
          <td className="user-link">
            <RouteLink to={`/account-info/${from}`}>
              {from}
            </RouteLink>
          </td>
          <td>
            <IconPaymentConvertArrow />
          </td>
          <td className="user-link">
            <RouteLink to={`/account-info/${to}`}>
              {to}
            </RouteLink>
          </td>
          <td>
            <div className="date">{toStandardFormat(timestamp, LOCALE_KEYS.RU_RU)}</div>
          </td>
          <td className="digit-value amount">
            {amount}
          </td>
        </tr>
        <ModalWindow
          id={trxId}
          bodyClassName="transaction-details"
          size="lg"
        >
          <h2 className="mb-3">
            <FormattedMessage id="explorer.tr-details.title" />
          </h2>
          <Table>
            <tbody>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.date" /></td>
                <td>{toStandardFormat(timestamp, LOCALE_KEYS.RU_RU)}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.amount" /></td>
                <td>{amount}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.hash" /></td>
                <td>{trxId}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.to" /></td>
                <td>{to}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.memo" /></td>
                <td>{memo}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.from" /></td>
                <td>{from}</td>
              </tr>
              <tr>
                <td><FormattedMessage id="explorer.tr-details.block" /></td>
                <td>{block}</td>
              </tr>
            </tbody>
          </Table>
        </ModalWindow>
      </Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  openDetailsModal: id => dispatch(openModal(id)),
});

export const Transaction = connect(null, mapDispatchToProps)(TransactionUI);
