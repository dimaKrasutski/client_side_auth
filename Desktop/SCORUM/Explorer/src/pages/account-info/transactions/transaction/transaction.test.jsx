import React from 'react';
import configureStore from 'redux-mock-store';
import { Transaction } from './transaction';
import { DEFAULT_LOCALE } from '../../../../constants';

const mockStore = configureStore([]);
const timestamp = '2018-03-07T15:42:39';
const testTransaction = {
  timestamp,
  index: 5,
  op: {
    amount: '100000 SCR',
    from: 'user1',
    to: 'user2',
  },
};

describe('<Transaction />', () => {
  beforeAll(() => {
    Date.now = jest.genMockFunction().mockReturnValue(timestamp);
  });

  test('should match snapshot', () => {
    const wrapper =
      shallow(<Transaction store={mockStore({})} transaction={testTransaction} locale={DEFAULT_LOCALE} />);
    expect(wrapper).toMatchSnapshot();
  });
});
