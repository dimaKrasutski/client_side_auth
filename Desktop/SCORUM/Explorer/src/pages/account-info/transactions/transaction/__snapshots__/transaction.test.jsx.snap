// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transaction /> should match snapshot 1`] = `
<TransactionUI
  locale="en-us"
  openDetailsModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  transaction={
    Object {
      "index": 5,
      "op": Object {
        "amount": "100000 SCR",
        "from": "user1",
        "to": "user2",
      },
      "timestamp": "2018-03-07T15:42:39",
    }
  }
/>
`;
